### PR TITLE
Add failing test for broker url priority ref issue #4284

### PR DESCRIPTION
**Pull request by Checkroth**
## Description
This PR is to assist in showing the issue described here: https://github.com/celery/celery/issues/4284

All this PR does is add a failing test that _should_ be passing, if the issue above is resolved.

This PR does not solve the issue mentioned. I leave that up to the discretion of a more knowledgeable party, if they decide that it is indeed an issue at all. I do not expect this PR to be merged unless the issue is resolved and you want this test to remain in the repository.

### CODE REVIEW

File: t/unit/app/test_utils.py [hunk 1]
```diff
@@ -26,6 +26,13 @@ def test_find_value_for_key(self):
         assert self.app.conf.find_value_for_key(
             'always_eager') is False
 
+    def test_broker_priority(self, monkeypatch):
+        prepatch_broker_url = 'memory://thisshouldbethebroker'
+        self.app.conf.broker_url = prepatch_broker_url
+        monkeypatch.setenv('CELERY_BROKER_URL',
+                           'memory://thisshouldntbethebroker')
+        assert self.app.conf.broker_url == prepatch_broker_url
```
Comment (georgepsarakis) [inline t/unit/app/test_utils.py]: The application configuration has already been resolved at this stage, so I think this test case does not actually prove the issue. Configuration cannot be changed dynamically.

Comment (Checkroth) [reply t/unit/app/test_utils.py]: I got that feeling when I wrote this, however calling `self.app.conf.broker_url` after setting it to `prepatch_broker_url`, but before setting `CELERY_BROKER_URL` in the environment, it does return the `prepatch_broker_url` string

**Comment by georgepsarakis** [comment]
@Checkroth @auvipy I think the configuration works as expected. Please consider the following:

```
import os
from celery.app.utils import Settings  # sub-class of celery.utils.collections.ConfigurationView
configuration = Settings(dict(broker_url='amqp://hello.world'))
print configuration.broker_url  # amqp://hello.world
os.environ['CELERY_BROKER_URL'] = 'amqp://hello.world.from.environment.variable'
print configuration.broker_url  # amqp://hello.world.from.environment.variable
```

I hope the priority is clearer now.

However your test seems to raise another question regarding the behavior of `ConfigurationView` and properties in sub-classes such as `Settings.broker_url`; an exception should have been raised when you try to set a value for `broker_url` and this does not happen. Perhaps `AttributeDictMixin` is responsible for this.

**Comment by Checkroth** [comment]
@georgepsarakis Thank you for clarifying.

My initial issue was that, when explicitly setting your Celery config, the broker url would read from your environment variables rather than your explicit setting if `CELERY_BROKER_URL` was set.

It appears that this is by design, so this pull request adds no value to Celery. I will write my thoughts on the issue over at #4284 , because I still feel that there is something to be done to avoid this confusion in the future.

**Pull request closed without merging**